src/internal/connector/discovery/api/sites.go

```diff
@@ -85,15 +87,67 @@ func (c Sites) GetAll(ctx context.Context, errs *fault.Bus) ([]models.Siteable,
 	return us, el.Failure()
 }
 
+const uuidRE = "[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}"
+
+// matches a site ID, with or without a doman name.  Ex, either one of:
+// 10rqc2.sharepoint.com,deadbeef-0000-0000-0000-000000000000,beefdead-0000-0000-0000-000000000000
+// deadbeef-0000-0000-0000-000000000000,beefdead-0000-0000-0000-000000000000
+var siteIDRE = regexp.MustCompile("(.+,)?" + uuidRE + "," + uuidRE)
+
+const webURLGetTemplate = "https://graph.microsoft.com/v1.0/sites/%s:/%s"
+
+// GetByID looks up the site matching the given ID.  The ID can be either a
+// canonical site id or a webURL.  Assumes the webURL is complete and well formed;
+// eg: https://10rqc2.sharepoint.com/sites/Example
 func (c Sites) GetByID(ctx context.Context, id string) (models.Siteable, error) {
-	resp, err := c.stable.Client().SitesById(id).Get(ctx, nil)
-	if err != nil {
-		return nil, graph.Wrap(ctx, err, "getting site")
+	var (
+		resp models.Siteable
+		err  error
+	)
+
+	ctx = clues.Add(ctx, "given_site_id", id)
+
+	if siteIDRE.MatchString(id) {
+		resp, err = c.stable.Client().SitesById(id).Get(ctx, nil)
+		if err != nil {
+			return nil, graph.Wrap(ctx, err, "getting site by id")
+		}
+	} else {
+		var (
+			url   = strings.TrimPrefix(id, "https://")
+			parts = strings.SplitN(url, "/", 1)
+			host  = parts[0]
+			path  string
+		)
+
+		if len(parts) > 1 {
+			path = parts[1]
+		}
```

Comment:
this should probably just use the URL package in golang so we don't have to worry about parsing it ourselves and other weirdness that can cause

parsing URLs: https://pkg.go.dev/net/url#Parse
info in a URL: https://pkg.go.dev/net/url#URL

Reply:
Huh, the funny thing is I have a different PR where I did just that.  Wonder why my brain skipped out on me here?